tests/metrics/functional/test_classification.py

```diff
@@ -346,6 +347,52 @@ def test_auroc(pred, target, expected):
     assert score == expected
 
 
+def test_multiclass_auroc():
+    with pytest.raises(ValueError,
+                       match="Multiclass AUROC metric expects the target scores to be "
+                             "probabilities, i.e. they should sum up to 1.0 over classes"):
+        _ = multiclass_auroc(pred=torch.tensor([[0.9, 0.9],
+                                                [1.0, 0]]),
+                             target=torch.tensor([0, 1]))
+
+    with pytest.raises(ValueError,
+                       match=r"Number of classes found in in 'target' (.*) "
+                             r"does not equal the number of columns in 'pred' (.*). "
+                             r"Multiclass AUROC is not defined when all of the classes do not "
+                             r"occur in the target labels."):
+        _ = multiclass_auroc(pred=torch.rand((4, 3)).softmax(dim=1),
+                             target=torch.tensor([1, 0, 1, 0]))
+
+    with pytest.raises(ValueError,
+                       match=r"Number of classes deduced from 'pred' (.*) does not equal "
+                             r"the number of classes passed in 'num_classes' (.*)."):
+        _ = multiclass_auroc(pred=torch.rand((5, 4)).softmax(dim=1),
+                             target=torch.tensor([0, 1, 2, 2, 3]),
+                             num_classes=6)
+
+
+def test_multiclass_auroc_against_sklearn():
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+    n_samples = 300
+    for n_cls in [2, 5, 10, 50]:
```

Comment:
can we have this in parametrization so it easier to spot the issue...

Reply:
Done.